doc/user/methods/random_forest_regressor.md

```diff
@@ -0,0 +1,639 @@
+  // should be ignored (e.g. the responses are not weighted).
+  template<bool UseWeights, typename ResponsesType, typename WeightsType>
+  double OutputLeafValue(const ResponsesType& responses,
+                         const WeightsType& weights);
+};
+```
+
+***Note:*** this API differs from the `FitnessFunction` API required for
+[`DecisionTree`](decision_tree.md)! <!-- TODO: fix link! -->
+
+---
+
+#### `DimensionSelectionType`
+
+ * When splitting a decision tree, `DimensionSelectionType` proposes possible
+   dimensions to try splitting on.
+ * `MultipleRandomDimensionSelect` _(default)_ is available for drop-in usage
+   and proposes a different random subset of dimensions at each decision tree
+   node.
+    - By default each random subset is of size `sqrt(d)` where `d` is the number
+      of dimensions in the data.
+    - If constructed as `MultipleRandomDimensionSelect(n)` and passed to the
+      constructor of `RandomForestRegressor` or the `Train()` function, each 
+      random subset will be of size `n`.
+ * Each `RandomForestRegressor` [constructor](#constructors) and each version of
+   the [`Train()`](#training) function optionally accept an instantiated
+   `DimensionSelectionType` object as the very last parameter (after
+   `maxDepth` in the constructor, or `warmStart` in `Train()`), in case some
+   internal state in the dimension selection mechanism is required.
+ * A custom class must implement three simple functions:
+
+
+```c++
+class CustomDimensionSelect
+{
+ public:
+  // Get the first dimension to try.
+  // This should return a value between `0` and `data.n_rows`.
+  size_t Begin();
+
+  // Get the next dimension to try.  Note that internal state can be used to
+  // track which candidate dimension is currently being looked at.
+  // This should return a value between `0` and `data.n_rows`.
+  size_t Next();
+
+  // Get a value indicating that all dimensions have been tried.
+  size_t End() const;
+
+  // The usage pattern of `DimensionSelectionType` by `DecisionTree` is as
+  // follows, assuming that `dim` is an instantiated `DimensionSelectionType`
+  // object:
+  //
+  // for (size_t dim = dim.Begin(); dim != dim.End(); dim = dim.Next())
+  // {
+  //   // ... try to split on dimension `dim` ...
+  // }
+};
+```
+
+---
+
+#### `NumericSplitType`
+
+ * Specifies the strategy to be used during training when splitting a numeric
+   feature.
+ * The `BestBinaryNumericSplit` _(default)_ class is available for drop-in
+   usage and finds the best binary (two-way) split among all possible binary
+   splits.
+ * The `RandomBinaryNumericSplit` class is available for drop-in usage and
+   will select a split randomly between the minimum and maximum values of a
+   dimension.  It is very efficient but does not yield splits that maximize
+   the gain.  (Used by the `ExtraTrees` variant of
+   [`RandomForest`](random_forest.md).) <!-- TODO: fix link! -->
```

Comment:
I think linking to `random_forest.md#fully-custom-behavior` could work here, but even just to the regular random forest documentation page is okay too.  Up to you.